_episodes/04-use-license.md

```diff
@@ -5,7 +5,185 @@ exercises: 0
+>
+>    **Question:** are you allowed to use the photos that you downloaded for your machine learning algorithm?
+{: .challenge}
+
+### What a licence does 
+
+A licence is an official permission to do, use, or own something. Basically, a licence waives parts of a copyright by defining terms and conditions for the usage of a product.   
+A shorthand definition of license is "a promise by the licensor not to sue licensees" if they comply with the licence terms and conditions.   
+
+What an open source licence does?
+
+Authors of code tend to share it expecting that others will use, modify and share it. However, as the default is exclusive copyright, you need to explicitely state that you allow others to use, modify and share your code.   
+The way to do it, is by assigning an open source licence to the code that you share with others, even if you are not publishing it but simply sharing it, e.g., by email.  
+Notice that "if you do not assign an open source license to your project, everybody who contributes to it becomes an exclusive copyright holder of their work. That means 'nobody' can use, copy, distribute, or modify their contributions – includes yourself." [[ref]](https://opensource.guide/legal/)   
+
+Also, you should be aware of whether other people's code, that you want to reuse or include in your project, has a licence and what type of licence it has. If you use, modify, or distribute a non licenced code, you are potentially liable to be sued by the code's owner.
+
+How can you know which permissions you have from a licence? How can you know whether a project has an open source licence? See [https://choosealicense.com/](https://choosealicense.com/)
+
+> ### Challenge
+>
+> Work in pairs. Ask learners to say wich permissions they have from a licence of a given software. Write permissions on sticky notes and stick them under each software name. Instructors should push learners to be more creative in terms of permissions. For example, can you sell the software? Can you sell support?
+{: .challenge}
+
+### Different licenses for different types of products (even data)
+
+
+## Tell why is important that a product/code has a licence
+
+* Why you need a licence
+* Explain the differences between concepts such as [open](https://opensource.org/docs/osd) and proprietary
+
+## Tell what is the importance of third-party dependencies on your product/code
+
+Third-party dependencies are any library that you are using in your project that was wrote by someone else. For example, if your data analysis in R uses ggplot2 for the visualisation, ggplot2 would be a third-party dependence. Third-party dependencies have their own licenses that can be different from the license that you apply to your project but they should be compatible. The freedom provided by your license must not violate the restriction described in the license of the third-party depencency that you are using. For example:
```

Comment:
I think it's a little more subtle than this here. Licenses of dependencies will affect your project differently whether you need to include the code in your own project, or whether you are simply linking to it. There are probably very few cases where including the source code of ggplot2 inside another project would make sense, and I worry it will give the wrong impression to the learners. For instance, you could compare the [fs package](https://cran.r-project.org/web/packages/fs/index.html) which includes the libuv C++ library; and the [magick package](https://cran.r-project.org/web/packages/magick/index.html) which does not include the source code from Image Magick but links to it. In the first case, the fs package needs to be released with a GPL3 license because of the libuv package. In the second case, the magick packages is released with an MIT license even though ImageMagick is released under a custom GPL-compatible license.